### ISSUE TITLE
AWS Signature Version 4 Signing

### ISSUE THREAD

**Issue by maruks** [dakrone/clj-http#299]
please add AWS Signature Version 4 Signing
http://docs.aws.amazon.com/general/latest/gr/signature-v4-examples.html#signature-v4-examples-java

**Comment by dakrone** [comment]
I'm not sure what you're asking for? Are you asking for documentation? An example file in the `examples` directory?

**Comment by maruks** [comment]
It would be nice to have AWS authentication for.   

I would like to do something like
(client/post "http://aws.gateway" {:aws [key-id secret-key api-key]})
and have http request signed

**Comment by dakrone** [comment]
I agree with @RutledgePaulV, I think this is a little out of the scope for clj-http, would documenting that the library works be a good solution for now @maruks?

**Comment by dakrone** [comment]
I pushed 36bdc14498516af16639cc7a5f0c181dbed4a448 related to this
